### PR TITLE
feat: add timezone on warehouse session

### PR DESCRIPTION
**Pull request by rephus**
Closes: #9720

### Description:

Screenshot from 2024-04-12 01-44-57

### Things not done in this PR 

- [ ] Sharing URLs : If I (Person A) share a URL to a chart that isn’t saved with someone (Person B) in a different timezone, the chart should show in Person A’s timezone.
- [ ] explore from here? does this take into consideration the timezone from the chart?: It uses the timezone from the chart, not your local timezone.
- [ ] Bigquery timezone conversion

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by rephus** [comment]
Bigquery doesn't support tiemzone with sessions, a date conversion is required, it will be done on a separate PR

File: packages/common/src/types/metricQuery.ts [hunk 1]
```diff
@@ -143,4 +144,5 @@ export type MetricQueryRequest = {
     customDimensions?: CustomDimension[];
     granularity?: DateGranularity;
     metadata?: MetricQuery['metadata'];
+    timezone?: string;
```
Comment (magnew) [inline packages/common/src/types/metricQuery.ts:147]: I'm planning to make this an Enum with the picker checkin. Any concerns?

**Review by magnew — COMMENTED** [review]
Looks good!

**Review by magnew — APPROVED** [review]

File: packages/frontend/src/hooks/useQueryResults.tsx [hunk 2]
```diff
@@ -82,6 +83,7 @@ const getQueryResults = async ({
     const timezoneFixQuery = query && {
```
Comment (magnew) [inline packages/frontend/src/hooks/useQueryResults.tsx:83]: There are some other places where we do something similar to this. I'm just noting that we should update those too (ex useCompliedSQL, useCalculateTotals)

File: packages/warehouses/.eslintrc.js [hunk 3]
```diff
@@ -17,5 +17,6 @@ module.exports = {
         'no-case-declarations': 'off',
         'import/prefer-default-export': 'off',
         'class-methods-use-this': 'off',
+        'no-console': 'off',
```
[the remark this answers is not in the document]
Comment (rephus) [reply packages/warehouses/.eslintrc.js:20]: no, we decided to remove console log warnings on warehouses

File: packages/common/src/types/metricQuery.ts (hunk 1 above)
Comment (rephus) [reply packages/common/src/types/metricQuery.ts:147]: yeah, that makes sense, I didn't know what values to give, so I used a generic string

**Pull request merged**